### ISSUE TITLE
No such property BUILD_FILE_NAMES

### ISSUE THREAD

**Issue by ruuda** [thought-machine/please#379]
Consider the following minimal `BUILD`:

```
new_http_archive(
    name = 'ghc_archive',
    urls = ['https://downloads.haskell.org/~ghc/8.2.2/ghc-8.2.2-x86_64-deb8-linux.tar.xz'],
    hashes = ['sha1: 6a74b608bbe658d87c428304a42873cd2ad5f337'],
    build_file = "ghc.BUILD",
)
```

When I then run `plz build --nocolour` I get the following error:
```
22:11:38.344   ERROR: //:all failed:
Config has no such property BUILD_FILE_NAMES
Build stopped after 50ms. 1 target failed:
    //:all
src/parse/rules/subrepo_rules.build_defs, line 97, column 45: error: Config has no such property BUILD_FILE_NAMES
    if build_file:
        cmd += ' && mv $SRCS_BUILD $OUT/' + CONFIG.BUILD_FILE_NAMES[0]
                                            ^
    elif build_file_content:

Traceback:
src/parse/rules/subrepo_rules.build_defs, line 97, column 45: cmd += ' && mv $SRCS_BUILD $OUT/' + CONFIG.BUILD_FILE_NAMES[0]

src/parse/rules/subrepo_rules.build_defs, line 96: 5:       if build_file:
                                   BUILD: 1: 1:   new_http_archive(
```
This is using Please 13.0.3.

**Comment by ruuda** [comment]
I was able to work around this by replacing the following line
https://github.com/thought-machine/please/blob/435d5d19aedb177a506db0d3d36f1f76c6774a23/src/parse/rules/subrepo_rules.build_defs#L97
with just 
```
cmd += ' && mv $SRCS_BUILD $OUT/BUILD'
```

Setting `BuildFileName = BUILD` in my `.plzconfig` did not help.

**Comment by peterebden** [comment]
Ah yeah, spotted that the other day. Fix is in #382, if that proves controversial or slow I'll break the fix out separately.
